network/p2p/cache/node_blocklist_wrapper_test.go

```diff
@@ -151,7 +152,32 @@ func (s *NodeBlocklistWrapperTestSuite) TestBlacklistedNode() {
 		for _, i := range combinedIdentities {
 			require.False(s.T(), i.Ejected) // Ejected flag should still have the original value (false here)
 		}
+	})
+
+	// this tests the edge case where the  Identities func is invoked with the p2p.NotEjectedFilter. Block listed
+	// nodes are expected to be filtered from the identity list returned after setting the ejected field.
+	s.Run("Identities(p2p.NotEjectedFilter) should not return block listed nodes", func() {
+		blocklistLookup := blocklist.Lookup()
+		honestIdentities := unittest.IdentityListFixture(8)
+		combinedIdentities := honestIdentities.Union(blocklist)
+		combinedIdentities = combinedIdentities.DeterministicShuffle(1234)
+		numIdentities := len(combinedIdentities)
+
+		s.provider.On("Identities", mock.Anything).Return(combinedIdentities)
+
+		identities := s.wrapper.Identities(p2p.NotEjectedFilter)
 
+		require.Equal(s.T(), len(honestIdentities), len(identities)) // expected only honest nodes to be returned
+		for _, i := range identities {
+			_, isBlocked := blocklistLookup[i.NodeID]
+			require.Equal(s.T(), isBlocked, i.Ejected)
```

Comment:
nit: make this check more explicit
```suggestion
			_, isBlocked := blocklistLookup[i.NodeID]
			require.False(s.T(), isBlocked)
			require.False(s.T(), i.Ejected)
```

Reply:
https://github.com/onflow/flow-go/pull/3748/commits/2ae1c8b903737e765c6c9423800824db5ad7506b